geoutils/georaster.py

```diff
@@ -775,31 +775,46 @@ def intersection(self, rst):
         else:
             return extent
 
-    def show(self, **kwargs):
+    def show(self, band=None, **kwargs):
         """ Show/display the image, with axes in projection of image.
 
         This method is a wrapper to rasterio.plot.show. Any **kwargs which you give
         this method will be passed to rasterio.plot.show.
 
-        You can also pass in **kwargs to be used by the underlying imshow or 
+        :param band: which band to plot, from 0 to self.count-1 (default is all)
+        :type band: int
+
+        :returns: None
+        :rtype: None
+
+        You can also pass in **kwargs to be used by the underlying imshow or
         contour methods of matplotlib. The example below shows provision of
         a kwarg for rasterio.plot.show, and a kwarg for matplotlib as well::
 
             import matplotlib.pyplot as plt
             ax1 = plt.subplot(111)
             mpl_kws = {'cmap':'seismic'}
             myimage.show(ax=ax1, mpl_kws)
-
-        :returns: None
-        :rtype: None
-
         """
         # If data is not loaded, need to load it
         if not self.isLoaded:
             self.load()
 
+        # Check if specific band selected, or take all
+        # rshow takes care of image dimensions
```

Comment:
that's pretty neat

Reply:
did you test it?! it would be good to put an example in pytest to ensure it does not fail; there's no need for an assertion statement, the fact that it would run without failing is enough information for this, I think

Reply:
I'll look into it

Reply:
I just did! You can try it in the new tests (better if you set DO_PLOT=True).
I saw people adding "assert True" at teh end of their test, but not sure if that's really useful...